From f799060ff9c063777095e56d522f3025543ae8a5 Mon Sep 17 00:00:00 2001
From: z00512904 <z00512904@huawei.com>
Date: Mon, 22 Aug 2022 03:29:44 +0000
Subject: [PATCH] [Huawei]: add oom notifier call for oom panic
 
Offering: iAOS 3.2
[IMPACT] : add oom notifier call for oom panic
note: update glibc
 
Signed-off-by: z00512904 <z00512904@huawei.com>
---
 mm/oom_kill.c | 1 +
 1 file changed, 1 insertion(+)
 
diff --git a/mm/oom_kill.c b/mm/oom_kill.c
index 2933e2beb..73ab33e34 100644
--- a/mm/oom_kill.c
+++ b/mm/oom_kill.c
@@ -1090,6 +1090,7 @@ static void check_panic_on_oom(struct oom_control *oc)
 	if (is_sysrq_oom(oc))
 		return;
 	dump_header(oc, NULL);
+	oom_type_notifier_call(0, oc);
 	panic("Out of memory: %s panic_on_oom is enabled\n",
 		sysctl_panic_on_oom == 2 ? "compulsory" : "system-wide");
 }
-- 
2.17.1
 
